app/views/contexts/calculated_summary_context.py

```diff
@@ -298,7 +298,16 @@ def _get_answer_format(self, groups: Iterable[Mapping]) -> Tuple[dict, list]:
                             "unit": answer.get("unit"),
                             "unit_length": answer.get("unit_length"),
                             "currency": answer.get("currency"),
+                            "decimal_places": 0,
                         }
+
+                        if (
+                            answer.get("decimal_places")
```

Comment:
It could be done without extra nesting if you keep the ifs combined
```
if (decimal_places := answer.get("decimal_places")) and decimal_places > answer_format["decimal_places"]:
  answer_format["decimal_places"] = decimal_places
```
however, I think the function is clear too, resolving 👍